contracts/halo-pool/src/formulas.rs

```diff
@@ -1,57 +1,61 @@
-use crate::state::PoolInfo;
-use cosmwasm_std::Uint128;
+use cosmwasm_std::{Uint128, Decimal};
 
-pub fn calc_reward(pool_info: &PoolInfo, current_time: u64) -> Uint128 {
-    let _reward_per_second = pool_info.reward_per_second;
-    let start_time = pool_info.start_time;
-    let end_time = pool_info.end_time;
-
-    if current_time < start_time {
-        return Uint128::zero();
+/// Returns the multiplier over the given _from_ and _to_ range.
+/// The multiplier is zero if the _to_ range is before the _end_.
+/// The multiplier is the _end_ minus _from_ if the _from_ range is after the _end_.
+/// Otherwise, the multiplier is the _to_ minus _from_.
+pub fn get_multiplier(from: u64, to: u64, end: u64) -> u64 {
+    if to < end {
```

Comment:
to < from thì sao

Reply:
Case này thì ko sảy ra đc vì `from` luôn là `last_reward_time` luôn nhỏ hơn `to` luôn luôn là `current time` ạ, hàm này đc dùng trong update_pool và query_pending reward ạ.
Nếu để cho sau này con dùng nữa thì em có thể add cho case đó return về 0 ạ